webauthn-authenticator-rs/src/win10/gui.rs

```diff
@@ -0,0 +1,271 @@
+//! GUI code for Windows WebAuthn API.
+//!
+//! Windows WebAuthn API needs a [HWND] parameter to know where to attach its
+//! UI to, but the `webauthn-authenticator-rs` API doesn't know anything about
+//! [HWND]s.
+//!
+//! Also, when running in a console application, getting a _functional_ [HWND]
+//! is a broken mess, because [GetConsoleWindow] only works properly for
+//! "classical" `cmd.exe` prompts: Windows Terminal
+//! [has focus issues, even though the issue is "closed"][terminal], and VS
+//! Code [also has z-order issues][vscode]. Both make for an awful experience,
+//! and [the normal work-arounds][hack] don't work when a console app can't
+//! set the window title verbatim (such as in VS Code).
+//!
+//! There is some prior art in [windows-fido-bridge], but this ended up with
+//! a distinct design that doesn't hard-code work-arounds for single apps.
+//!
+//! **Warning:** This hasn't yet been tested in a Windows GUI application,
+//! and is subject to change based on user feedback.
+//!
+//! [Window] creates a 1x1 pixel window for Windows WebAuthn API to use:
```

Comment:
I wonder if there's any security tools that flag a 1x1 window being popped like that 🤔 time will tell, I guess 😃

Reply:
🤷